app/components/ChangeColorInput.jsx

```diff
@@ -0,0 +1,61 @@
+import React, { Component, PropTypes } from 'react';
+import BaseComponent from './BaseComponent';
+import { CompactPicker } from 'react-color';
+
+export default class ChangeColorInput extends BaseComponent {
+
+  constructor() {
+    super();
+    this.state = {
+      displayColorPicker: false,
+      color: "#cccccc"
+ 	 },
+    this.handleClick = this.handleClick.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+    this.handleValueChange = this.handleValueChange.bind(this);
+    this.handleClearClick = this.handleClearClick.bind(this);
+    this.onChange = this.onChange.bind(this);
```

Comment:
BaseComponent has a bindAll() method for this very purpose. in fact that's the only thing BaseComponent is good for!